domains/pallets/evm-tracker/src/lib.rs

```diff
@@ -76,15 +98,89 @@ mod pallet {
     impl<T: Config> Pallet<T> {
         /// Replace ContractCreationAllowedBy setting in storage, as a domain sudo call.
         #[pallet::call_index(0)]
-        #[pallet::weight(<T as frame_system::Config>::DbWeight::get().reads_writes(0, 1))]
+        #[pallet::weight(<T as frame_system::Config>::DbWeight::get().reads_writes(1, 1))]
         pub fn set_contract_creation_allowed_by(
```

Comment:
One entrance for setting the allowlist should be enough, `set_contract_creation_allowed_by` seems duplicated now as there is also `inherent_set_contract_creation_allowed_by `.

If we also want to allow the consensus sudo to set the allowlist then we can simply use `ensure_signed_or_root` in the `send_evm_domain_set_contract_creation_allowed_by_call` call of `pallet-domains`, in this way there is only one flow to set the allowlist: `send_evm_domain_set_contract_creation_allowed_by_call -> inherent_set_contract_creation_allowed_by` and we don't need to change anything about the domain sudo.

Reply:
Once we remove the `set_contract_creation_allowed_by` call, the `IsPrivateEvmDomain` state can also be removed since all the necessary checks will be done in the consensus chain, and for a public EVM the `ContractCreationAllowedBy` should always be a default value that can be used directly in `is_private_evm_domain/is_allowed_to_create_contracts/is_allowed_to_create_unsigned_contracts`

Reply:
Thanks, this is really useful feedback! I wish I'd got it before I started this change, it would have saved me a lot of work.